### PR TITLE
Refactoring Kubelet's cAdvisor interface into a package.

### PR DESCRIPTION
**Pull request by vmarmol**
This will make it easier to start running the real cAdvisor alongside
Kubelet. This change is primarily no-op refactoring. The main behavioral
change is that we always create a cAdvisor interface and expect it to
always be available. When we make a request, if cAdvisor is not
connected the request fails with a connection error. This failure is
handled today as well.

### CODE REVIEW

**Comment by vmarmol** [comment]
e2e tests pass

File: pkg/kubelet/cadvisor/cadvisor.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package cadvisor
+
+import (
+	"github.com/google/cadvisor/client"
+	"github.com/google/cadvisor/info"
+)
+
+type cadvisorClient struct {
+	client *client.Client
```
Comment (xiang90) [inline pkg/kubelet/cadvisor/cadvisor.go]: if we make client an embedded filed, we do not even need to repeat the method.
```
type cadvisorClient struct{
    *client.Client
}
```

Comment (vmarmol) [inline pkg/kubelet/cadvisor/cadvisor.go]: Neat! Done and thanks for the suggestion :)

File: pkg/kubelet/kubelet_test.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func init() {
 	util.ReallyCrash = true
 }
 
-func newTestKubelet(t *testing.T) (*Kubelet, *dockertools.FakeDockerClient, *sync.WaitGroup) {
+func newTestKubelet(t *testing.T) (*Kubelet, *dockertools.FakeDockerClient, *sync.WaitGroup, *cadvisorMock) {
```
Comment (vishh) [inline pkg/kubelet/kubelet_test.go]: nit: Can we return a struct that includes all the current return types instead?

File: pkg/kubelet/kubelet_test.go [hunk 3]
```diff
@@ -1172,7 +1174,7 @@ func TestFieldPath(t *testing.T) {
 	}
 }
 
-type mockCadvisorClient struct {
+type cadvisorMock struct {
```
Comment (vishh) [inline pkg/kubelet/kubelet_test.go]: Why not place the mock in a separate file?

**Comment by vishh** [comment]
@vmarmol: This PR needs a rebase. 
Just a couple of nits. Otherwise LGTM.

**Comment by vishh** [comment]
@vmarmol:  This needs a rebase.

**Comment by vishh** [comment]
My bad. I didn't realize that I haven't merged this PR and I merged your recent one. :(

File: pkg/kubelet/kubelet_test.go (hunk 3 above)
Comment (vmarmol) [inline pkg/kubelet/kubelet_test.go]: Done.

File: pkg/kubelet/kubelet_test.go (hunk 2 above)
Comment (vmarmol) [inline pkg/kubelet/kubelet_test.go]: This is probably OK for now, I added a TODO in case it grows more or changes in use.

**Comment by vmarmol** [comment]
Thanks for the review @vishh! Rebased and pushed.

**Comment by vishh** [comment]
LGTM. 
Can be merged once the CI tests pass.

**Pull request merged**